client/packages/invoices/src/Prescriptions/DetailView/Payments/PaymentsModal.tsx

```diff
@@ -54,15 +52,19 @@ export const PaymentsModal: FC<PaymentsModalProps> = ({
       insurance.isActive && !DateUtils.isExpired(insurance.expiryDate)
   );
 
-  const selectedInsurance = insuranceData?.find(
-    ({ insuranceProviders }) => insuranceProviders?.id === insuranceId
-  );
+  const selectedInsurance = insuranceData?.find(({ id }) => id === insuranceId);
 
   const { plugins } = usePluginProvider();
   const pluginEvents = usePluginEvents({
     isDirty: false,
   });
 
+  const totalAfterTax = prescriptionData?.pricing.totalAfterTax ?? 0;
+  const discountPercentage = selectedInsurance?.discountPercentage ?? 0;
+  const totalToBePaidByInsurance =
+    totalAfterTax * ((selectedInsurance?.discountPercentage ?? 0) / 100);
+  const totalToBePaidByPatient = totalAfterTax - totalToBePaidByInsurance;
```

Comment:
The value `totalToBePaidByPatient` is *only* used by the plugin, so wouldn't it be better to calculate this value inside the plugin?

Reply:
I wasn't sure what else might be affecting the price, so I'd rather the App told the plugin what it thought the correct amount is, and then the plugin could do something different with it as needed?
Feels like the right thing to me...